src/simulator/main_utils.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+#include "../common/AlgorithmRegistrar.h"
+#include "my_simulator.h"
+#include "robot.h"
+#include <dlfcn.h>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+struct SimInfo {
+    MySimulator simulator;
+    std::string house_file_name;
+    std::string house_output_path;
+};
+
+struct AlgoInfo {
+    std::unique_ptr<AbstractAlgorithm> algorithm;
```

Comment:
why do we need algo_file_path?

Reply:
Thought we will need but we dont, will remove it